### ISSUE TITLE
[Bug]: Search timeout after load, flush and load in test case "test_search_load_flush_load"

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#15628]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: dc65b1ae
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): 2.0.1.dev3
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Search timeout reporting "Deadline Exceeded"

```
[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-873)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-874)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-875)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-876)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-877)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-878)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-879)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-880)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-881)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-882)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-883)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-884)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-885)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-886)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-887)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-888)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-889)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-890)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-891)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-892)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-893)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-894)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-895)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-896)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-897)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-898)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-899)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-900)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-901)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-902)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-903)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-904)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-905)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-906)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-907)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-908)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-909)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-910)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-911)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-912)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-913)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-914)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-915)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-916)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-917)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-918)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-919)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-920)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-921)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-922)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-923)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-924)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-925)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-926)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-927)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-928)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-929)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-930)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-931)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-932)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-933)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-934)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-935)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-936)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-937)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-938)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-939)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-940)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-941)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-942)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-943)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-944)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-945)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-946)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-947)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-948)[](https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/#step-153-log-949)[2022-02-17T17:29:47.724Z] self = <test_search_20.TestCollectionSearch object at 0x7ff7d53e7550>, nb = 5000

[2022-02-17T17:29:47.724Z] nq = 500, dim = 8, auto_id = True, _async = True

[2022-02-17T17:29:47.724Z] 

[2022-02-17T17:29:47.724Z]     @pytest.mark.tags(CaseLabel.L1)

[2022-02-17T17:29:47.724Z]     def test_search_load_flush_load(self, nb, nq, dim, auto_id, _async):

[2022-02-17T17:29:47.724Z]         """

[2022-02-17T17:29:47.724Z]         target: test search when load before flush

[2022-02-17T17:29:47.724Z]         method: 1. insert data and load

[2022-02-17T17:29:47.724Z]                 2. flush, and load

[2022-02-17T17:29:47.724Z]                 3. search the collection

[2022-02-17T17:29:47.724Z]         expected: search success with limit(topK)

[2022-02-17T17:29:47.724Z]         """

[2022-02-17T17:29:47.724Z]         # 1. initialize with data

[2022-02-17T17:29:47.724Z]         collection_w = self.init_collection_general(prefix, auto_id=auto_id, dim=dim)[0]

[2022-02-17T17:29:47.724Z]         # 2. insert data

[2022-02-17T17:29:47.724Z]         insert_ids = cf.insert_data(collection_w, nb, auto_id=auto_id, dim=dim)[3]

[2022-02-17T17:29:47.724Z]         # 3. load data

[2022-02-17T17:29:47.724Z]         collection_w.load()

[2022-02-17T17:29:47.724Z]         # 4. flush and load

[2022-02-17T17:29:47.724Z]         collection_w.num_entities

[2022-02-17T17:29:47.724Z]         collection_w.load()

[2022-02-17T17:29:47.724Z]         # 5. search

[2022-02-17T17:29:47.724Z]         vectors = [[random.random() for _ in range(dim)] for _ in range(nq)]

[2022-02-17T17:29:47.724Z]         collection_w.search(vectors[:nq], default_search_field,

[2022-02-17T17:29:47.724Z]                             default_search_params, default_limit,

[2022-02-17T17:29:47.724Z]                             default_search_exp, _async=_async,

[2022-02-17T17:29:47.724Z]                             check_task=CheckTasks.check_search_results,

[2022-02-17T17:29:47.724Z]                             check_items={"nq": nq,

[2022-02-17T17:29:47.724Z]                                          "ids": insert_ids,

[2022-02-17T17:29:47.724Z]                                          "limit": default_limit,

[2022-02-17T17:29:47.724Z] >                                        "_async": _async})

[2022-02-17T17:29:47.724Z] 

[2022-02-17T17:29:47.724Z] testcases/test_search_20.py:1091: 

[2022-02-17T17:29:47.724Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2022-02-17T17:29:47.724Z] base/collection_wrapper.py:122: in search

[2022-02-17T17:29:47.724Z]     timeout=timeout, **kwargs).run()

[2022-02-17T17:29:47.724Z] check/func_check.py:54: in run

[2022-02-17T17:29:47.724Z]     result = self.check_search_results(self.response, self.func_name, self.check_items)

[2022-02-17T17:29:47.724Z] check/func_check.py:198: in check_search_results

[2022-02-17T17:29:47.724Z]     search_res = search_res.result()

[2022-02-17T17:29:47.725Z] /usr/local/lib/python3.6/site-packages/pymilvus/orm/future.py:30: in result

[2022-02-17T17:29:47.725Z]     return self.on_response(self._f.result())

[2022-02-17T17:29:47.725Z] /usr/local/lib/python3.6/site-packages/pymilvus/client/asynch.py:175: in result

[2022-02-17T17:29:47.725Z]     self.exception()

[2022-02-17T17:29:47.725Z] /usr/local/lib/python3.6/site-packages/pymilvus/client/asynch.py:232: in exception

[2022-02-17T17:29:47.725Z]     raise self._exception

[2022-02-17T17:29:47.725Z] /usr/local/lib/python3.6/site-packages/pymilvus/client/asynch.py:219: in done

[2022-02-17T17:29:47.725Z]     self._response.append(future.result(timeout=None))

[2022-02-17T17:29:47.725Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2022-02-17T17:29:47.725Z] 

[2022-02-17T17:29:47.725Z] self = <_MultiThreadedRendezvous of RPC that terminated with:

[2022-02-17T17:29:47.725Z] 	status = StatusCode.DEADLINE_EXCEEDED

[2022-02-17T17:29:47.725Z] 	details = "Deadline Exc...37:19530","file":"src/core/lib/surface/call.cc","file_line":1068,"grpc_message":"Deadline Exceeded","grpc_status":4}"

[2022-02-17T17:29:47.725Z] >

[2022-02-17T17:29:47.725Z] timeout = None

[2022-02-17T17:29:47.725Z] 

[2022-02-17T17:29:47.725Z]     def result(self, timeout=None):

[2022-02-17T17:29:47.725Z]         """Returns the result of the computation or raises its exception.

[2022-02-17T17:29:47.725Z]     

[2022-02-17T17:29:47.725Z]         See grpc.Future.result for the full API contract.

[2022-02-17T17:29:47.725Z]         """

[2022-02-17T17:29:47.725Z]         with self._state.condition:

[2022-02-17T17:29:47.725Z]             timed_out = _common.wait(self._state.condition.wait,

[2022-02-17T17:29:47.725Z]                                      self._is_complete,

[2022-02-17T17:29:47.725Z]                                      timeout=timeout)

[2022-02-17T17:29:47.725Z]             if timed_out:

[2022-02-17T17:29:47.725Z]                 raise grpc.FutureTimeoutError()

[2022-02-17T17:29:47.725Z]             else:

[2022-02-17T17:29:47.725Z]                 if self._state.code is grpc.StatusCode.OK:

[2022-02-17T17:29:47.725Z]                     return self._state.response

[2022-02-17T17:29:47.725Z]                 elif self._state.cancelled:

[2022-02-17T17:29:47.725Z]                     raise grpc.FutureCancelledError()

[2022-02-17T17:29:47.725Z]                 else:

[2022-02-17T17:29:47.725Z] >                   raise self

[2022-02-17T17:29:47.725Z] E                   grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:

[2022-02-17T17:29:47.725Z] E                   	status = StatusCode.DEADLINE_EXCEEDED

[2022-02-17T17:29:47.725Z] E                   	details = "Deadline Exceeded"

[2022-02-17T17:29:47.725Z] E                   	debug_error_string = "{"created":"@1645115437.564185137","description":"Error received from peer ipv4:10.201.71.137:19530","file":"src/core/lib/surface/call.cc","file_line":1068,"grpc_message":"Deadline Exceeded","grpc_status":4}"
```

### Expected Behavior

Search successfully

### Steps To Reproduce

```markdown
Running nightly:
https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/pipeline/129/

Test case:

    def test_search_load_flush_load(self, nb, nq, dim, auto_id, _async):
        """
        target: test search when load before flush
        method: 1. insert data and load
                2. flush, and load
                3. search the collection
        expected: search success with limit(topK)
        """
        # 1. initialize with data
        collection_w = self.init_collection_general(prefix, auto_id=auto_id, dim=dim)[0]
        # 2. insert data
        insert_ids = cf.insert_data(collection_w, nb, auto_id=auto_id, dim=dim)[3]
        # 3. load data
        collection_w.load()
        # 4. flush and load
        collection_w.num_entities
        collection_w.load()
        # 5. search
        vectors = [[random.random() for _ in range(dim)] for _ in range(nq)]
        collection_w.search(vectors[:nq], default_search_field,
                            default_search_params, default_limit,
                            default_search_exp, _async=_async,
                            check_task=CheckTasks.check_search_results,
                            check_items={"nq": nq,
                                         "ids": insert_ids,
                                         "limit": default_limit,
                                         "_async": _async})
```
```


### Anything else?

Analysi info:

1. Log:
https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/weekly-milvus-nightly-ci%2Fweekly-milvus-ci-test/detail/weekly-milvus-ci-test/21/artifacts
"milvus-distributed-21-pymilvus-e2e-logs.tar.gz)"

2. Case failed timeline:
[2022-02-17T16:30:39.241Z] [gw3] [ 59%] FAILED testcases/test_search_20.py::TestCollectionSearch::test_search_load_flush_load[5000-500-8-True-True] 

3. collection name: search_collection_wNYWBRyH

**Comment by xige-16** [comment]
on time1, querynode 1 has sealed segment 1, and querynode 2 has growing segment 2, all node's global sealed segments = [s1]
on time 2, querynode 1 and 2 receive the search request,  querynode2 execute search and return result which has searched sealed segments= [s1], global sealed segments=[s1], but querynode 1 push search request to unsolved buffer when service time is smaller than search request's timestamp.
on time3, querynode2 load sealed segment 2, and all querynode's global sealed segments=[s1, s2]
on time4, querynode1 execute search request and return searched sealed segment = [s1], global sealed segments=[s1, s2]
on time5, proxy get all search result, but find search sealed segments=[s1], global sealed segments=[s1, s2], The proxy continues to wait for the query to segment2, and finally the search times out.

**Comment by xige-16** [comment]
I will fix this bug, you can set the consistency of the query, choose eventual consistency to avoid this problem. @binbinlv

**Comment by czs007** [comment]
> 
@xige-16 @binbinlv 
Is this bug being fixed? Time to catch up with the 2.0.2 release?

**Comment by binbinlv** [comment]
@congqixia @xige-16 
How about the fix for this issue?

Thanks.
